hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -47,10 +42,10 @@ public HiveClientPool(int poolSize, Configuration conf) {
   }
 
   @Override
-  protected HiveMetaStoreClient newClient()  {
+  protected IMetaStoreClient newClient()  {
     try {
       try {
-        return CLIENT_CTOR.newInstance(hiveConf);
+        return RetryingMetaStoreClient.getProxy(hiveConf, null, HiveMetaStoreClient.class.getName());
```

Comment:
Maybe I'm mistaken, but this is the CTOR that is @VisibleForTesting?

Can we use: 

``` 
public static IMetaStoreClient getProxy(
      Configuration hiveConf, boolean allowEmbedded)
```